dlt/common/storages/configuration.py

```diff
@@ -145,6 +145,8 @@ class FilesystemConfiguration(BaseConfiguration):
     kwargs: Optional[DictStrAny] = None
     client_kwargs: Optional[DictStrAny] = None
     deltalake_storage_options: Optional[DictStrAny] = None
+    enable_state_cleanup: bool = True
```

Comment:
This should be a config on the filesystem destination as it is not applied anywhere else. I would not have the "enable" flag, but interpret max_state_files=None as this feature being disabled.